core/src/main/scala/com/banno/kafka/publish/PublishStreamOps.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Jack Henry & Associates, Inc.®
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.banno.kafka.publish
+
+import cats.Applicative
+import cats.effect.Concurrent
+import fs2.Stream
+
+final class PublishStreamOps[F[_], A](private val stream: Stream[F, F[A]])
+    extends AnyVal {
+
+  def batched(implicit A: Applicative[F]): Stream[F, A] =
+    stream.evalMapChunk(identity)
+
+  def batchedPar(maxConcurrent: Int)(implicit C: Concurrent[F]): Stream[F, A] =
+    if (maxConcurrent > 0) stream.parEvalMap(maxConcurrent)(identity)
+    else stream.parEvalMapUnbounded(identity)
```

Comment:
If I accidentally call `batchedPar(-1)` accidentally, I will still process the stream, but crucially *not* in a batched/chunked fashion, right? This is a thing that feels like it can catch people off guard and is tricky to debug. Would it be better if we returned a `Option[Stream[F, A]]` to be explicit about that?

```scala
def batchedPar(maxConcurrent: Int)(implicit C: Concurrent[F]): Stream[F, A] =
  (maxConcurrent > 0)
    .guard[Option]
    .as(stream.parEvalMap(maxConcurrent)(identity))
```

Reply:
If we really want to guard against that I would push us to use refinement types to ensure that the int is positive

Reply:
On second thought the refined type is a little overkill and annoying to expose through api, so that mixed with the fact that I cant actually think of a solid reason to need to limit maxConcurrent for our use case I'd sooner throw the whole method away than try to make it bulletproof

Reply:
Sorry, I was missing the forrest for the trees. I conflated the `stream.evalMapChunk` bit above to imply that we wanted to explicitly chunk things in all of these functions. So ignore my original comment.